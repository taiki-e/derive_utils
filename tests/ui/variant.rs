// SPDX-License-Identifier: Apache-2.0 OR MIT

use example_derive::Iterator;

#[derive(Iterator)]
enum EnumWZV {} //~ ERROR may not be used on enums without variants

#[derive(Iterator)]
enum EnumWD {
    A = 2, //~ ERROR may not be used on enums with discriminants
    B,
}

#[derive(Iterator)]
enum EnumWZF1<B> {
    A, //~ ERROR may not be used on enums with variants with zero fields
    B(B),
}

#[derive(Iterator)]
enum EnumWZF2<B> {
    A(), //~ ERROR may not be used on enums with variants with zero fields
    B(B),
}

#[derive(Iterator)]
enum EnumWMF<A, B> {
    A(A),
    B(A, B), //~ ERROR may not be used on enums with variants with multiple fields
}

#[derive(Iterator)]
enum EnumWNF<A, B> {
    A { x: A }, //~ ERROR may not be used on enums with variants with named fields
    B(B),
}

fn main() {}
