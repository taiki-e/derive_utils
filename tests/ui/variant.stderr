error: may not be used on enums without variants
 --> tests/ui/variant.rs:6:1
  |
6 | enum EnumWZV {} //~ ERROR may not be used on enums without variants
  | ^^^^^^^^^^^^^^^

error: may not be used on enums with discriminants
  --> tests/ui/variant.rs:10:9
   |
10 |     A = 2, //~ ERROR may not be used on enums with discriminants
   |         ^

error: may not be used on enums with variants with zero fields
  --> tests/ui/variant.rs:16:5
   |
16 |     A, //~ ERROR may not be used on enums with variants with zero fields
   |     ^

error: may not be used on enums with variants with zero fields
  --> tests/ui/variant.rs:22:5
   |
22 |     A(), //~ ERROR may not be used on enums with variants with zero fields
   |     ^^^

error: may not be used on enums with variants with multiple fields
  --> tests/ui/variant.rs:29:5
   |
29 |     B(A, B), //~ ERROR may not be used on enums with variants with multiple fields
   |     ^^^^^^^

error: may not be used on enums with variants with named fields
  --> tests/ui/variant.rs:34:5
   |
34 |     A { x: A }, //~ ERROR may not be used on enums with variants with named fields
   |     ^^^^^^^^^^
